account_mass_reconcile/models/advanced_reconciliation.py

```diff
@@ -100,5 +100,26 @@ def _opposite_matchers(move_line):
         :yield: matchers as tuple ('matcher key', value(s))
         """
         yield ('partner_id', move_line['partner_id'])
-        yield ('ref', ((move_line['ref'] or '').lower().strip(),
-                       move_line['name'].lower().strip()))
+        yield ('ref', ((move_line['name'] or '').lower().strip(),
+                       move_line['ref'].lower().strip()))
```

Comment:
FTR, I thought we should write `(move_line['ref'] or '')` because the ref is not required, but it can't happen here since `_skip_line` exclude lines without ref.

Reply:
Note: we should indeed check for empty ref because `_skip_line` exclude only the credit move line, not the lines to match.